src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -641,6 +659,10 @@ public static void WriteObjectInitialization(this CodeWriter writer, JsonObjectS
             {
                 writer.Line($"{objAdditionalProperties.Type} {dictionaryVariable:D} = new {objAdditionalProperties.Type}();");
             }
+            else if (serialization.ObjectType is not null && serialization.ObjectType.HasRawDataInHeirarchy)
+            {
+                writer.Line($"{Parameter.RawData.Type} rawData = new {Parameter.RawData.Type}();");
```

Comment:
Shouldn't the `rawData` be a variable? What if there is a property called `rawData`?

Reply:
Not sure I follow the question it is a `field` not a `property`

Reply:
You could try 
```
model Test {
  rawData: string;
}
```
and then the deserialize function will have two `rawData`
```C#
string rawData = default;
Dictionary<string, BinaryData> rawData = new Dictionary<string, BinaryData>();
```

Reply:
Gotcha, perhaps we call it `__rawData` and then name the stack variable `_rawDAta` so that the chance of conflict is much lower, nearly zero.

Reply:
Why not we just do something like:
```C#
var rawDataVariable = new CodeWriterDeclaration("rawData");
writer.Line($"{Parameter.RawData.Type} {rawDataVariable:D} = new {Parameter.RawData.Type}();");
```